### DISCUSSION TITLE
Is there a possibility to skip APPIUM_HOME validation at Appium server startup?

### DISCUSSION THREAD

**Discussion by samijt** [appium/appium#19248] [Ideas]
In one of the 2.0.0 beta releases `APPIUM_HOME ` validation added check for the `write` permission ([PR 18462](https://github.com/appium/appium/pull/18462)). I commented to that PR this same issue, but here it is now as a new idea.

The change broke our CI environemnt where we install Appium with higher privileges with drivers. Users are not supposed to update anything but just use the pre-installed Appium with the pre-installed drivers. Now Appium won't start as it requires write permission to `APPIUM_HOME` folder. This worked earlier (beta-62 and earlier) as expected and we did not have this problem.

What is the recommended way to install and setup Appium on multi-user environment and not having separate driver installations per user? From where I could read what Appium server writes to `APPIUM_HOME` at runtime? I'm assuming that this is needed for **driver and plugin management**, but not by Appium server itself, right?

**Comment by KazuCocoa** [comment]
So far, yes. Mostly APPIUM_HOME is to manage drivers/plugins, but potentially appium will write temporary files in there too.
e.g. https://github.com/appium/appium/tree/master/packages/strongbox (current the implementation depends on https://www.npmjs.com/package/env-paths, but I cannot guarantee it won't write in APPIUM_HOME for the future)

**Reply by KazuCocoa** [reply]
@jlipps Did we have something update...? https://github.com/appium/appium/pull/18462#issuecomment-1516438810

**Reply by jlipps** [reply]
Yeah some drivers might write into APPIUM_HOME maybe. I would say Appium is not designed for a use case where users are allowed to run the appium server but not allowed to run the driver/plugin update commands. Our view so far is that anyone running the appium server is a server administrator with full rights to run the CLI for any purpose. If you want users to simply consume Appium servers, maybe having running servers waiting for webdriver commands would be better than also allowing them to start servers? Or having an npm package which combines appium and drivers at certain versions which everyone on your team is required to use?
